.claude/scripts/loop-contract-check.sh

```
@@ -0,0 +1,169 @@
+GITIGNORE=".gitignore"
+
+fails=0
+pass() { printf '  ok  %s\n' "$1"; }
+fail() { printf '  FAIL %s\n' "$1" >&2; fails=$((fails + 1)); }
+have() { [ -f "$1" ] || { fail "missing file: $1"; return 1; }; }
+
+echo "loop-contract-check — verifying invariants across the loop artifacts"
+
+# --- 1. All artifacts exist -----------------------------------------------
+for f in "$SKILL" "$DOC" "$RUNNER" "$RESOLVER" "$REVIEWER" "$XAML_BINDER" "$LAUNCHER" "$GITIGNORE"; do
+  have "$f" && pass "exists: $f" || fail "exists: $f"
```

Shellcheck (0.11.0)

[info] Note that A && B || C is not if-then-else. C may run when A is true.

(SC2015)